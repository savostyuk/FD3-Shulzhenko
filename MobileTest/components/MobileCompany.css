.MobileCompany
{

}

.MobileCompanyName
{
    font-weight: bold;
    padding: 5px 0;
}

table.MobileCompanyClients
{
    border: 1px solid gray;
    border-collapse: collapse;
    width: 100%;
    margin: 10px 0;
    text-align: center;
}
tr.Headings {
    font-weight: bold;
}
tr.Headings td{
    padding: 5px 0;
}