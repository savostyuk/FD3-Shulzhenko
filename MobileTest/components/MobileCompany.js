import React from 'react';
import PropTypes from 'prop-types';

import {clientEvents} from './events';
import {formEvents} from './events';

import MobileClient from './MobileClient';
import ClientForm from './ClientForm';

import './MobileCompany.css';

class MobileCompany extends React.PureComponent {

    static propTypes = {
        clients:PropTypes.arrayOf(
            PropTypes.shape({
                id: PropTypes.number.isRequired,
                fam: PropTypes.string.isRequired,
                im: PropTypes.string.isRequired,
                otch: PropTypes.string.isRequired,
                balance: PropTypes.number.isRequired,
            })
        ),
    };

    state = {
        clients: this.props.clients,
        filtered: 0,   //0 начальное состояние+показать всех, 1- показать активных, 2 - показать заблокированных
        formMode: 0,  //форма закрыта, 1 - редактировать, 2 - добавить нового
        selectedClientCode: null,
    };

    allShow = () =>{
        this.setState({filtered: 0});
    }

    activeShow = () =>{
        this.setState({filtered: 1});
    }

    blockedShow = () =>{
        this.setState({filtered: 2});
    }

    filterClient = (mode) => {
        if (mode === 0) {
            return this.state.clients;
        }
        if (mode === 1) {
            return this.state.clients.filter (v => v.balance>=0);
        }
        if (mode === 2) {
            return this.state.clients.filter (v => v.balance<0);
        }

    }

    componentDidMount() {
        clientEvents.addListener('EEditClient', this.editedClient);
        clientEvents.addListener('EDeleteClient', this.deletedClient);
        formEvents.addListener('EAddNewClient', this.saveChanged);
        formEvents.addListener('ECancelForm', this.cancelChanged);
    }

    componentWillUnmount() {
        clientEvents.removeListener('EEditClient', this.editedClient);
        clientEvents.removeListener('EDeleteClient', this.deletedClient);
        formEvents.removeListener('EAddNewClient', this.saveChanged);
        formEvents.removeListener('ECancelForm', this.cancelChanged);
    }

    editedClient = (id) =>{
        this.setState({formMode: 1});
        this.setState({selectedClientCode: id});
    }

    deletedClient = (id) =>{
        var Question = confirm("Вы хотите удалить клиента?");
        if (Question) {
            var filterClient = this.state.clients.filter(v =>
                v.id !== id);
            this.setState({clients: filterClient});
        }
    }
    addClient = () =>{
        this.setState({formMode: 2});
    }
    saveChanged = (client) =>{
        if (this.state.formMode===1) {
            let indexEdit;
            this.state.clients.forEach((v, i) => {
                if (v.id===client.id) {
                    indexEdit=i;
                }
            });
            let editedClients = this.state.clients;
            editedClients[indexEdit] = client;
            this.setState({clients: editedClients});
        }

        if (this.state.formMode===2) {
            let maxId=0;
            this.state.clients.forEach(v => {
                if (v.id>maxId) {
                    maxId=v.id
                }
            })
            client.id = maxId+1;
            this.state.clients.push(client);
        }
        this.setState({formMode: 0});
    }


    cancelChanged = () =>{
        this.setState({formMode: 0});
    }

    render() {

        console.log("MobileCompany render");

        let filteredClient = this.filterClient(this.state.filtered);

        var clientsCode=filteredClient.map( client =>
            <MobileClient key={client.id} info={client}  />
        );

        let selectedClient;
        if (this.state.formMode===1){
            selectedClient = this.state.clients.filter (client => client.id===this.state.selectedClientCode);
        }

        return (
            <div className='MobileCompany'>
                <hr />
                    <input className="showAll" type="button" value="Все" onClick={this.allShow} />
                    <input className="showActive" type="button" value="Активные" onClick={this.activeShow} />
                    <input className="showBlocked" type="button" value="Заблокированные" onClick={this.blockedShow} />
                <hr />
                <table className='MobileCompanyClients'>
                    <tbody>
                    <tr className="Headings">
                        <td>Фамилия</td>
                        <td>Имя</td>
                        <td>Отчество</td>
                        <td>Баланс</td>
                        <td>Статус</td>
                        <td>Редактировать</td>
                        <td>Удалить</td>
                    </tr>
                    {clientsCode}
                    </tbody>
                </table>
                <input type="button" value="Добавить клиента" onClick={this.addClient} />
                {this.state.formMode===1 &&
                <ClientForm
                    key = {selectedClient[0].id}
                    info = {selectedClient}
                    formMode = {this.state.formMode}/>
                }
                {this.state.formMode===2 &&
                    <ClientForm formMode={this.state.formMode}/>
                }
            </div>
        )
            ;

    }

}

export default MobileCompany;
