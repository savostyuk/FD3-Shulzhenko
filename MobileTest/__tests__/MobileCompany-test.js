"use strict";

import React from 'react';
import renderer from 'react-test-renderer';

import MobileCompany from "../components/MobileCompany";

const clientsArr=[
    {id:101, fam:"Иванов", im:"Иван", otch:"Иванович", balance:200},
    {id:105, fam:"Сидоров", im:"Сидор", otch:"Сидорович", balance:250},
    {id:110, fam:"Петров", im:"Петр", otch:"Петрович", balance:180},
    {id:120, fam:"Григорьев", im:"Григорий", otch:"Григорьевич", balance:-220},
];

test ('Работа кнопки Все', () => {

    // создаём тестовую версию компонента
    const component = renderer.create(
        <MobileCompany clients={clientsArr}/>
    );

    // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    let componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    // найдём в вёрстке компонента саму кнопку
    const buttonElem = component.root.find( el => el.props.className==='showAll');
    // и "нажмём" на неё
    buttonElem.props.onClick();

    // получаем уже изменённый снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    // "нажмём" кнопку ещё раз
    buttonElem.props.onClick();

    // и получаем окончательный снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

});

test ('Работа кнопки Активные', () => {

    // создаём тестовую версию компонента
    const component = renderer.create(
        <MobileCompany clients={clientsArr}/>
    );

    // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    let componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    // найдём в вёрстке компонента саму кнопку
    const buttonElem = component.root.find( el => el.props.className==='showActive');
    // и "нажмём" на неё
    buttonElem.props.onClick();

    // получаем уже изменённый снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    // "нажмём" кнопку ещё раз
    buttonElem.props.onClick();

    // и получаем окончательный снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

});

test ('Работа кнопки Заблокированные', () => {

    // создаём тестовую версию компонента
    const component = renderer.create(
        <MobileCompany clients={clientsArr}/>
    );

    // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    let componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    // найдём в вёрстке компонента саму кнопку
    const buttonElem = component.root.find( el => el.props.className==='showBlocked');
    // и "нажмём" на неё
    buttonElem.props.onClick();

    // получаем уже изменённый снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    // "нажмём" кнопку ещё раз
    buttonElem.props.onClick();

    // и получаем окончательный снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

});