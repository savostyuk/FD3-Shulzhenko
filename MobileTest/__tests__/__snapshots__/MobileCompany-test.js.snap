// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Работа кнопки Активные 1`] = `
<div
  className="MobileCompany"
>
  <hr />
  <input
    className="showAll"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    className="showActive"
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    className="showBlocked"
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="MobileCompanyClients"
  >
    <tbody>
      <tr
        className="Headings"
      >
        <td>
          Фамилия
        </td>
        <td>
          Имя
        </td>
        <td>
          Отчество
        </td>
        <td>
          Баланс
        </td>
        <td>
          Статус
        </td>
        <td>
          Редактировать
        </td>
        <td>
          Удалить
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Иванов
        </td>
        <td>
          Иван
        </td>
        <td>
          Иванович
        </td>
        <td>
          200
        </td>
        <td
          className="statusactive"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Сидоров
        </td>
        <td>
          Сидор
        </td>
        <td>
          Сидорович
        </td>
        <td>
          250
        </td>
        <td
          className="statusactive"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Петров
        </td>
        <td>
          Петр
        </td>
        <td>
          Петрович
        </td>
        <td>
          180
        </td>
        <td
          className="statusactive"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="statusblocked"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`Работа кнопки Активные 2`] = `
<div
  className="MobileCompany"
>
  <hr />
  <input
    className="showAll"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    className="showActive"
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    className="showBlocked"
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="MobileCompanyClients"
  >
    <tbody>
      <tr
        className="Headings"
      >
        <td>
          Фамилия
        </td>
        <td>
          Имя
        </td>
        <td>
          Отчество
        </td>
        <td>
          Баланс
        </td>
        <td>
          Статус
        </td>
        <td>
          Редактировать
        </td>
        <td>
          Удалить
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Иванов
        </td>
        <td>
          Иван
        </td>
        <td>
          Иванович
        </td>
        <td>
          200
        </td>
        <td
          className="statusactive"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Сидоров
        </td>
        <td>
          Сидор
        </td>
        <td>
          Сидорович
        </td>
        <td>
          250
        </td>
        <td
          className="statusactive"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Петров
        </td>
        <td>
          Петр
        </td>
        <td>
          Петрович
        </td>
        <td>
          180
        </td>
        <td
          className="statusactive"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`Работа кнопки Активные 3`] = `
<div
  className="MobileCompany"
>
  <hr />
  <input
    className="showAll"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    className="showActive"
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    className="showBlocked"
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="MobileCompanyClients"
  >
    <tbody>
      <tr
        className="Headings"
      >
        <td>
          Фамилия
        </td>
        <td>
          Имя
        </td>
        <td>
          Отчество
        </td>
        <td>
          Баланс
        </td>
        <td>
          Статус
        </td>
        <td>
          Редактировать
        </td>
        <td>
          Удалить
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Иванов
        </td>
        <td>
          Иван
        </td>
        <td>
          Иванович
        </td>
        <td>
          200
        </td>
        <td
          className="statusactive"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Сидоров
        </td>
        <td>
          Сидор
        </td>
        <td>
          Сидорович
        </td>
        <td>
          250
        </td>
        <td
          className="statusactive"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Петров
        </td>
        <td>
          Петр
        </td>
        <td>
          Петрович
        </td>
        <td>
          180
        </td>
        <td
          className="statusactive"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`Работа кнопки Все 1`] = `
<div
  className="MobileCompany"
>
  <hr />
  <input
    className="showAll"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    className="showActive"
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    className="showBlocked"
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="MobileCompanyClients"
  >
    <tbody>
      <tr
        className="Headings"
      >
        <td>
          Фамилия
        </td>
        <td>
          Имя
        </td>
        <td>
          Отчество
        </td>
        <td>
          Баланс
        </td>
        <td>
          Статус
        </td>
        <td>
          Редактировать
        </td>
        <td>
          Удалить
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Иванов
        </td>
        <td>
          Иван
        </td>
        <td>
          Иванович
        </td>
        <td>
          200
        </td>
        <td
          className="statusactive"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Сидоров
        </td>
        <td>
          Сидор
        </td>
        <td>
          Сидорович
        </td>
        <td>
          250
        </td>
        <td
          className="statusactive"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Петров
        </td>
        <td>
          Петр
        </td>
        <td>
          Петрович
        </td>
        <td>
          180
        </td>
        <td
          className="statusactive"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="statusblocked"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`Работа кнопки Все 2`] = `
<div
  className="MobileCompany"
>
  <hr />
  <input
    className="showAll"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    className="showActive"
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    className="showBlocked"
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="MobileCompanyClients"
  >
    <tbody>
      <tr
        className="Headings"
      >
        <td>
          Фамилия
        </td>
        <td>
          Имя
        </td>
        <td>
          Отчество
        </td>
        <td>
          Баланс
        </td>
        <td>
          Статус
        </td>
        <td>
          Редактировать
        </td>
        <td>
          Удалить
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Иванов
        </td>
        <td>
          Иван
        </td>
        <td>
          Иванович
        </td>
        <td>
          200
        </td>
        <td
          className="statusactive"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Сидоров
        </td>
        <td>
          Сидор
        </td>
        <td>
          Сидорович
        </td>
        <td>
          250
        </td>
        <td
          className="statusactive"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Петров
        </td>
        <td>
          Петр
        </td>
        <td>
          Петрович
        </td>
        <td>
          180
        </td>
        <td
          className="statusactive"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="statusblocked"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`Работа кнопки Все 3`] = `
<div
  className="MobileCompany"
>
  <hr />
  <input
    className="showAll"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    className="showActive"
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    className="showBlocked"
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="MobileCompanyClients"
  >
    <tbody>
      <tr
        className="Headings"
      >
        <td>
          Фамилия
        </td>
        <td>
          Имя
        </td>
        <td>
          Отчество
        </td>
        <td>
          Баланс
        </td>
        <td>
          Статус
        </td>
        <td>
          Редактировать
        </td>
        <td>
          Удалить
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Иванов
        </td>
        <td>
          Иван
        </td>
        <td>
          Иванович
        </td>
        <td>
          200
        </td>
        <td
          className="statusactive"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Сидоров
        </td>
        <td>
          Сидор
        </td>
        <td>
          Сидорович
        </td>
        <td>
          250
        </td>
        <td
          className="statusactive"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Петров
        </td>
        <td>
          Петр
        </td>
        <td>
          Петрович
        </td>
        <td>
          180
        </td>
        <td
          className="statusactive"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="statusblocked"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`Работа кнопки Заблокированные 1`] = `
<div
  className="MobileCompany"
>
  <hr />
  <input
    className="showAll"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    className="showActive"
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    className="showBlocked"
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="MobileCompanyClients"
  >
    <tbody>
      <tr
        className="Headings"
      >
        <td>
          Фамилия
        </td>
        <td>
          Имя
        </td>
        <td>
          Отчество
        </td>
        <td>
          Баланс
        </td>
        <td>
          Статус
        </td>
        <td>
          Редактировать
        </td>
        <td>
          Удалить
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Иванов
        </td>
        <td>
          Иван
        </td>
        <td>
          Иванович
        </td>
        <td>
          200
        </td>
        <td
          className="statusactive"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Сидоров
        </td>
        <td>
          Сидор
        </td>
        <td>
          Сидорович
        </td>
        <td>
          250
        </td>
        <td
          className="statusactive"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Петров
        </td>
        <td>
          Петр
        </td>
        <td>
          Петрович
        </td>
        <td>
          180
        </td>
        <td
          className="statusactive"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="statusblocked"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`Работа кнопки Заблокированные 2`] = `
<div
  className="MobileCompany"
>
  <hr />
  <input
    className="showAll"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    className="showActive"
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    className="showBlocked"
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="MobileCompanyClients"
  >
    <tbody>
      <tr
        className="Headings"
      >
        <td>
          Фамилия
        </td>
        <td>
          Имя
        </td>
        <td>
          Отчество
        </td>
        <td>
          Баланс
        </td>
        <td>
          Статус
        </td>
        <td>
          Редактировать
        </td>
        <td>
          Удалить
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="statusblocked"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`Работа кнопки Заблокированные 3`] = `
<div
  className="MobileCompany"
>
  <hr />
  <input
    className="showAll"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    className="showActive"
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    className="showBlocked"
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="MobileCompanyClients"
  >
    <tbody>
      <tr
        className="Headings"
      >
        <td>
          Фамилия
        </td>
        <td>
          Имя
        </td>
        <td>
          Отчество
        </td>
        <td>
          Баланс
        </td>
        <td>
          Статус
        </td>
        <td>
          Редактировать
        </td>
        <td>
          Удалить
        </td>
      </tr>
      <tr
        className="MobileClientInfo"
      >
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="statusblocked"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;
